components/apimgt/org.wso2.carbon.apimgt.keymgt/src/test/java/org/wso2/carbon/apimgt/keymgt/handlers/DefaultKeyValidationHandlerTest.java

```diff
@@ -0,0 +1,153 @@
+        api.addResource(urlMapping);
+
+        Map<String, API> apiMap = new HashMap<>();
+        String key = API_CONTEXT + ":" + API_VERSION;
+        apiMap.put(key, api);
+
+        APIKeyValidationInfoDTO dto = new APIKeyValidationInfoDTO();
+        dto.setSubscriber(SUBSCRIBER);
+        dto.setApplicationName(APPLICATION_NAME);
+        dto.setApplicationId(APPLICATION_ID);
+        dto.setApplicationTier(TIER);
+        Set<String> scopeSet = new HashSet<>();
+        scopeSet.add(SCOPES);
+        dto.setScopes(scopeSet);
+        dto.setSubscriberTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        dto.setUserType(APIConstants.ACCESS_TOKEN_USER_TYPE_APPLICATION);
+
+        // TokenValidationContext for not default API
+        TokenValidationContext param1 = new TokenValidationContext();
+        param1.setValidationInfoDTO(dto);
+        param1.setContext(API_CONTEXT);
+        param1.setVersion(API_VERSION);
+        param1.setAccessToken(ACCESS_TOKEN);
+        param1.setMatchingResource(RESOURCE);
+        param1.setHttpVerb(HTTP_VERB);
+
+        // TokenValidationContext for default API version
+        TokenValidationContext param2 = new TokenValidationContext();
+        param2.setValidationInfoDTO(dto);
+        param2.setContext(API_CONTEXT);
+        param2.setVersion(DEFAULT_API_VERSION);
+        param2.setAccessToken(ACCESS_TOKEN);
+        param2.setMatchingResource(RESOURCE);
+        param2.setHttpVerb(HTTP_VERB);
+
+
+        SubscriptionDataHolder subscriptionDataHolder = Mockito.mock(SubscriptionDataHolder.class);
+        SubscriptionDataStore tenantSubscriptionStore = Mockito.mock(SubscriptionDataStore.class);
+        PrivilegedCarbonContext privilegedCarbonContext = Mockito.mock(PrivilegedCarbonContext.class);
+
+        PowerMockito.mockStatic(SubscriptionDataHolder.class);
+        Mockito.when(SubscriptionDataHolder.getInstance()).thenReturn(subscriptionDataHolder);
+
+        PowerMockito.mockStatic(PrivilegedCarbonContext.class);
+        Mockito.when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(privilegedCarbonContext);
+        Mockito.when(privilegedCarbonContext.getTenantDomain()).thenReturn(TENANT_DOMAIN);
+
+        Mockito.when(subscriptionDataHolder.getTenantSubscriptionStore(eq(TENANT_DOMAIN))).thenReturn(tenantSubscriptionStore);
+        Mockito.when(tenantSubscriptionStore.getApiByContextAndVersion(eq(API_CONTEXT), eq(API_VERSION))).thenReturn(api);
+
+
+        DefaultKeyValidationHandler defaultKeyValidationHandler = new DefaultKeyValidationHandler();
+
+        boolean isScopeValidated = defaultKeyValidationHandler.validateScopes(param1);
+        boolean isScopeValidated_default = defaultKeyValidationHandler.validateScopes(param2);
+
+        Assert.assertTrue("Scope validation fails for NOT default API!", isScopeValidated);
+        Assert.assertTrue("Scope validation fails for default API!", isScopeValidated_default);
```

Comment:
Let's change this message to "Scope validation fails for default API " + API_NAME